modulus/datapipes/climate/dummy.py

```diff
@@ -0,0 +1,185 @@
+        num_steps: int,
+        device: str = "cuda",
+        grid_size: Tuple[int, int] = (721, 1440),
+        base_temp: float = 15,
+        amplitude: float = 10,
+        noise_level: float = 2,
+        **kwargs: Any,
+    ):
+        self.num_days: int = num_samples_per_year
+        self.num_steps: int = num_steps
+        self.num_channels: int = len(channels)
+        self.device = device
+        self.grid_size: Tuple[int, int] = grid_size
+        start_time = time.time()
+        self.temperatures: np.ndarray = self.generate_data(
+            self.num_days,
+            self.num_channels,
+            self.grid_size,
+            base_temp,
+            amplitude,
+            noise_level,
+        )
+        print(
+            f"Generated synthetic temperature data in {time.time() - start_time:.2f} seconds."
+        )
+        self.extra_args: Dict[str, Any] = kwargs
+
+    def generate_data(
+        self,
+        num_days: int,
+        num_channels: int,
+        grid_size: Tuple[int, int],
+        base_temp: float,
+        amplitude: float,
+        noise_level: float,
+    ) -> np.ndarray:
+        """
+        Generates synthetic temperature data over a specified number of days for multiple atmospheric layers.
+
+        Args:
+            num_days (int): Number of days to generate data for.
+            num_channels (int): Number of channels representing different layers.
+            grid_size (tuple): Grid size (latitude, longitude).
+            base_temp (float): Base mean temperature for the data.
+            amplitude (float): Amplitude of temperature variations.
+            noise_level (float): Noise level to add stochasticity to the temperature.
+
+        Returns:
+            numpy.ndarray: A 4D array of temperature values across days, channels, latitudes, and longitudes.
+        """
+        days: np.ndarray = np.arange(num_days)
+        latitudes, longitudes = grid_size
+        daily_temps: np.ndarray = np.zeros(
+            (num_days, num_channels, latitudes, longitudes)
+        )
+
+        for day in days:
+            for channel in range(num_channels):
+                altitude_effect: float = (
+                    channel * -0.5
+                )  # Temperature decreases with altitude
+                lat_variation: np.ndarray = np.linspace(
+                    -amplitude, amplitude, latitudes
+                )
+                daily_temps[day, channel] = (
+                    base_temp
+                    + altitude_effect
+                    + np.outer(lat_variation, np.sin(2 * np.pi * day / 365))
+                    + np.random.normal(scale=noise_level, size=(latitudes, longitudes))
+                )
+        return daily_temps
+
+    def __len__(self) -> int:
+        """
+        Returns the number of samples available in the dataset.
+        """
+        return self.num_days - self.num_steps
+
+    def __getitem__(self, idx: int) -> torch.Tensor:
```

Comment:
This needs a test since it is a utility in the main repo

Reply:
Added tests